_posts/en/newsletters/2021-03-10-newsletter.md

```diff
@@ -158,7 +158,11 @@ BOLTs][bolts repo].*
 
 - [BOLTs #839][] Add 2016 blocks channel funding timeout (#839) FIXME:dongcarl
 
-- [BTCPay Server #2181][] Reenabling uppercase BECH32 in QR codes FIXME:Xekyo
+- [BTCPay Server #2181][] uppercases bech32 addresses when presenting Bitcoin
+  URIs as QR codes. This results in [less dense QR codes][bech32 uppercase qr]
+  as strings starting or ending with uppercase substrings can be encoded more
```

Comment:
Mh, given that they still use `bitcoin:` as the prefix, maybe the claim should be attenuated a bit.

Reply:
QR codes have a "mixed" mode where they can shift from one encoding to another multiple times at a cost per shift of about 20 bits, so the more contiguous uppercase characters you can add, the better, no matter where they occur in the string.  See "However, QR codes can support mixed character sets [...]" in https://bitcoinops.org/en/bech32-sending-support/#creating-more-efficient-qr-codes-with-bech32-addresses

Reply:
Oh, cool, I understood that as only working for sequences at the start or end.